crates/biome_js_analyze/src/semantic_analyzers/nursery/no_duplicate_else_if.rs

```diff
@@ -0,0 +1,208 @@
+use crate::semantic_services::Semantic;
+use biome_analyze::{context::RuleContext, declare_rule, Rule, RuleDiagnostic};
+use biome_console::markup;
+use biome_js_syntax::{AnyJsExpression, JsIfStatement, JsLogicalOperator, JsSyntaxKind};
+use biome_rowan::{AstNode, SyntaxNodeCast};
+
+fn flatten_parent(node: &AnyJsExpression) -> AnyJsExpression {
+    let expression_inside_parent = node
+        .as_js_parenthesized_expression()
+        .and_then(|parent| parent.expression().ok());
+
+    if let Some(expression) = expression_inside_parent {
+        return expression;
+    }
+    node.clone()
+}
+
+fn split_by_logical_operator(
+    operator: JsLogicalOperator,
+    node: &AnyJsExpression,
+) -> Vec<AnyJsExpression> {
```

Comment:
This function makes many `Vec` allocations. Instead of returning a `Vec`, I suggest passing a mutable reference to a `Vec`:

```suggestion
fn split_by_logical_operator(
    operator: JsLogicalOperator,
    node: &AnyJsExpression,
    result: &mut Vec<AnyJsExpression>,
) {
```